pkg/diagnose/check.go

```diff
@@ -24,10 +26,92 @@ func init() {
 	diagnosis.Register("check-datadog", diagnose)
 }
 
+func diagnose(diagCfg diagnosis.Config) []diagnosis.Diagnosis {
+	if diagCfg.RunningInAgentProcess {
+		return diagnoseInAgentProcess(diagCfg)
+	}
+
+	return diagnoseInCLIProcess(diagCfg)
+}
+
+func partitionChecks(checks []check.Check) [][]check.Check {
+	instancesMap := map[string][]check.Check{}
+
+	// partition per check (instances of a check groupped by check/integration)
+	for _, c := range checks {
+		checkName := c.String()
+		instances, exists := instancesMap[checkName]
+		if exists {
+			instances = append(instances, c)
+		} else {
+			instances = make([]check.Check, 1)
+			instances[0] = c
+		}
+		instancesMap[checkName] = instances
+	}
+
+	// copy map to array
+	instancesArr := make([][]check.Check, 0, len(instancesMap))
+	for _, instances := range instancesMap {
+		instancesArr = append(instancesArr, instances)
+	}
+
+	// sort this array
+	sort.Slice(instancesArr, func(i, j int) bool {
+		return instancesArr[i][0].String() < instancesArr[j][0].String()
+	})
+
+	return instancesArr
+}
+
+func getInstanceDiagnoses(instance check.Check) []diagnosis.Diagnosis {
+
+	// Get diagnoses
+	diagnoses, err := instance.GetDiagnoses()
+	if err == nil && len(diagnoses) > 0 {
+		// Set category as check name if it was not set
+		for i, d := range diagnoses {
+			if len(d.Category) == 0 {
+				diagnoses[i].Category = instance.String()
+			}
+		}
+	} else if err != nil {
+		// Check Diagnose method return error
+		diagnoses = append(diagnoses, diagnosis.Diagnosis{
+			Result:    diagnosis.DiagnosisUnexpectedError,
+			Name:      string(instance.ID()),
+			Category:  instance.String(),
+			Diagnosis: "Check Dianose failes with unexpected errors",
+			RawError:  err,
+		})
+	}
```

Comment:
Please rewrite to follow the regular pattern: 
```
if err != nil {
 // handle error
 return ...
}

// normal flow
return ...
```

Reply:
Changed (I actually did not know about existence of this pattern). Thank you for letting me know. See new commit.